File: src/files/FileJFTPFile.java
package files;

import jftp.JFTP.JFTPFile;

public class FileJFTPFile implements FileCommon {

	private final JFTPFile file;
	
	public FileJFTPFile(JFTPFile file) {
		this.file = file;
	}
	@Override
	public String getFileName() {
		return file.getName();
	}

	@Override
	public long getFileSize() {
		return Long.parseLong(file.getSize().trim());
	}

	@Override
	public boolean isFile() {
		return file.isFile();
	}

	@Override
	public boolean isDirectory() {
		return file.isDirectory();
	}

}

File: src/files/FileJFTPFile.kt
package files

import jftp.JFTP.JFTPFile

class FileJFTPFile(private val file: JFTPFile) : FileCommon {
    override fun getFileName(): String {
        return file.name
    }

    override fun getFileSize(): Long {
        return file.size.trim { it <= ' ' }.toLong()
    }

    override fun isFile(): Boolean {
        return file.isFile
    }

    override fun isDirectory(): Boolean {
        return file.isDirectory
    }
}
